### PR TITLE
codeintel: Expose inference results via GraphQL

### LINKED ISSUE #44120: Add documentation for auto-inference
Right now, when you navigate to the Inference page, all you see is:

> Code graph inference script
>
> Lua script that emits complete and/or partial auto-indexing job specifications. 

Would be helpful to link to the docs here once docs are added.

### PR DESCRIPTION
**Pull request by efritz**
Towards a fix for #44120 and #47647. This PR does the relevant backend work.

## Test plan

Tested by hand.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.84 kb) | 0.02% (+1.84 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ef+tr/codeintel-inference-embetterment/compare-report.html;tab=live_object) for a full comparison between the commits dc90ce6 and 93bb511 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by efritz** [comment]
@umpox Would it be possible to hand this PR over to you? Feel free to merge off of it, use this one as you please, destroy it, set fire to it, etc. I'm using the current override to test the inference page (it doesn't do anything special though):

```lua
local path = require("path")
local pattern = require("sg.autoindex.patterns")
local recognizer = require("sg.autoindex.recognizer")

local custom_recognizer = recognizer.new_path_recognizer {
    patterns = { pattern.new_path_basename("sg-test") },

    -- Invoked when go.mod files exist
    generate = function(_, paths)
        local jobs = {}
        for i = 1, #paths do
            table.insert(jobs, {
                steps = {},
                root = path.dirname(paths[i]),
                indexer = "test-override",
                indexer_args = {},
                outfile = "",
            })
        end

        return jobs
    end,
}

return require("sg.autoindex.config").new({
    ["custom.test"] = custom_recognizer,
})
```

Currently this PR exposes new GraphQL endpoints to run the inference script against a repository. Right now that repository is hard-coded just to get something working end-to-end.

image

Right now the modifications to the inference page is crufty since I hope we'll be replacing monaco. I didn't know how to hook into the event system here well enough to do it on change/save, so it only works on page load with the current contents of the editor.

Additionally, we should abstract the "index job description" component from the precise index timeline component so we can re-use it. In another ticket I mentioned that the current timeline view (only) is problematic because if it doesn't run you can't see it.

image

File: client/web/src/enterprise/codeintel/configuration/hooks/useInferJobs.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        key
+        command
+        startTime
+        exitCode
+        out
+        durationMilliseconds
+    }
+`
+
+export const useInferJobs = ({
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/configuration/hooks/useInferJobs.tsx:66]: cc @umpox You can use this hook for #47647.

Comment (umpox) [reply client/web/src/enterprise/codeintel/configuration/hooks/useInferJobs.tsx:66]: ty!

**Comment by Strum355** [comment]
Im wondering whether we should have a default text skeleton, like here https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/codeintel/autoindexing/internal/inference/service_generator_test.go?L28

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**